api/src/main/java/org/apache/iceberg/view/SQLViewRepresentation.java

```diff
@@ -36,17 +38,21 @@ default Type type() {
   String dialect();
 
   /** The default catalog when the view is created. */
+  @Nullable
   String defaultCatalog();
 
   /** The default namespace when the view is created. */
+  @Nullable
   Namespace defaultNamespace();
 
-  /** The query output schema at version create time, without aliases. */
-  Schema schema();
+  /** The query output schema id at version create time */
+  int schemaId();
```

Comment:
Since we have the flexibility to change this API until core implementation is complete, I changed the API to schemaID instead of schema. This is mostly because it simplifies the parsing logic and then it's still easy for a caller to obtain the schema from the top level view schema mapping.

Currently in the view spec, the schema ID is stored per SQL view representation. So in the current implementation building the representation when parsing is straightforward. 

There are a few routes to go here:

1.) Maintain schema ID in metadata and the API just returns schema ID as well (keeping parsing logic simple). That's what's done in this PR and I think is preferable since looking up schema via view.schemas().get(schemaID) should be straightforward.

2.) Preserve the schema at the API level, maintain schema ID in metadata. This complicates parsing logic (although not too much) because during parsing we need to pass the top level schemas list to SQLViewRepresentationParser https://iceberg.apache.org/view-spec/#view-metadata and then obtain the schema based on the parsed schema ID.

3.) Update the spec so that the entire schema object is stored in metadata and then serialize/deserialize.

Another topic (regardless of option 1 or 3) is the spec currently marks schemaID as optional for sql view representation. I think it must be required (I can't think of a case where for a SQL representation we don't want to maintain a well defined Iceberg schema, engines can still choose to ignore it if they want to although I can't really think of such a case). I can raise a PR to update that if we think it's the right approach. 

cc: @jzhuge @rdblue @jackye1995 @nastra

Reply:
I think there are still ways to use Schema, and ideally that is preferred from API perspective. For parser, similar to PartitionSpec parser, I would imagine the parser to take the schemas and use that. Is the same strategy achieveable here?

Reply:
Thanks @jackye1995 yeah that's achievable, that's approach 2 above. It does complicate parsing more than I'd like but it's very doable. I'll update the PR and the community can take a look and we can see which one we prefer more.

Reply:
After implementing it, while it's doable it does seem to complicate more. I'll see about simplifying it, but we may just want to go back to maintaining schema IDs or even serializing the entire schema to keep parsing really simple. and then engines can just do view.schema(repr.schemaId())

Reply:
I see. After taking a deeper look, there are multiple places that we have `schemaId()` as a part of the API model, such as in `Snapshot`. 

I was using the case of `PartitionSpec` to argue that we should stick with using `schema()`, but it was because there is a process of binding to a schema, and the serialized version of partition spec is unbounded.

So the question here is not which one is more convenient to implement parser, but if a view representation needs to bind to a schema at runtime, or have a static schema. I think the answer is that it is static, as described as "ID of the view’s schema when the version was created". So from that perspective, I agree `schemaId()` seems like a better choice so we don't need to cross reference existing schemas to make the parser work. 

Any thoughts? @amogh-jahagirdar @nastra

Reply:
@jackye1995 exactly my thoughts as well, a schema is known at the time of view creation so late binding like we do with PartitionSpec doesn't apply or would needlessly complicate the logic of constructing the representation. Will hold for @nastra and @rdblue thought as well

Reply:
Updated to use Integer schemaId(). Since schemaId may not be defined at creation time as discussed in https://github.com/apache/iceberg/pull/6611/files schemaId() can return null